source/material_model/visco_plastic.cc

```diff
@@ -929,6 +942,14 @@ namespace aspect
           frank_kamenetskii_rheology.initialize_simulator (this->get_simulator());
           frank_kamenetskii_rheology.parse_parameters(prm);
 
+          // Peierls creep parameters
+          use_peierls_creep = prm.get_bool ("Include Peierls creep");
+          if (use_peierls_creep)
+            {
+              peierls_creep.initialize_simulator (this->get_simulator());
+              peierls_creep.parse_parameters(prm);
+            }
```

Comment:
Can you use the same structure of a unique pointer as for Frank Kamenetskii in #3766?

Reply:
Actually, I don't think there is any need to use this check here and have removed it. `use_peierls_creep` is really only needed to decide about the compositing, which of course I forgot to add.